swaylock/render.c

```diff
@@ -18,7 +18,14 @@ static void set_color_for_state(cairo_t *cairo, struct swaylock_state *state,
 	} else if (state->auth_state == AUTH_STATE_CLEAR) {
 		cairo_set_source_u32(cairo, colorset->cleared);
 	} else {
-		cairo_set_source_u32(cairo, colorset->input);
+		if (state->xkb.caps_lock && state->args.show_caps_lock_indicator) {
+			cairo_set_source_u32(cairo, colorset->caps_lock);
+		} else {
+			if (state->xkb.caps_lock && !state->args.show_caps_lock_indicator) {
+				state->args.colors.text.input = state->args.colors.text.caps_lock;
```

Comment:
Also, can you please reset `state->args.color.text.input` after setting the cairo source?